### PR TITLE
Hot fix/fix paging in peoplewithbonusesfetchcontroller

### PR DESCRIPTION
**Pull request by Augustyniak**
Initially I just wanted to add tests for the `PeopleWithBonusesFetchController` class. During testing I found some bugs in it and decided to fix them.

Created tests aren't perfect - mostly because there contain some nasty casting - but I still think that this PR improves overall quality of the project.

### CODE REVIEW

File: GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift [hunk 1]
```diff
@@ -48,22 +50,26 @@ final class PeopleWithBonusesFetchController {
     }
 
     private func fetchPeopleWithBonuses(startingFromDate date: NSDate = NSDate.init(), fetchedBonuses: [Bonus], completionBlock: (Result<[Person]>) -> Void) {
-        let take = 100
-        let requestTemplate = TimestampableRequestTemplate<BonusesRequestTemplate>.init(requestTemplate: BonusesRequestTemplate(), date: date, take: take)
+        let requestTemplate = TimestampableRequestTemplate(requestTemplate: BonusesRequestTemplate(), date: date, take: self.pageSize)
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift]: Unneeded self here

File: GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift [hunk 2]
```diff
@@ -48,22 +50,26 @@ final class PeopleWithBonusesFetchController {
...
             case .Success(let bonuses):
-                var allBonuses: [Bonus] = fetchedBonuses.reverse()
+                var allBonuses: [Bonus] = fetchedBonuses
                 allBonuses.appendContentsOf(bonuses)
 
-                let people = allBonuses.uniqueReceivers(kPreferredPeopleCount)
-                if people.count >= kPreferredPeopleCount || bonuses.count < take {
+                let people = allBonuses.uniqueReceivers(strongSelf.preferredNumberOfPeople)
+
+                for person in people {
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift]: I am not sure which style we prefer : `for person in people` or `people.forEach` ?

File: GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift [hunk 3]
```diff
@@ -48,22 +50,26 @@ final class PeopleWithBonusesFetchController {
...
+                    print(person.id)
+                }
+
+                if people.count >= self?.preferredNumberOfPeople || bonuses.count < strongSelf.pageSize {
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift]: `self?` and `strongSelf` used on the same line

File: GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+struct TestRequestSender: RequestSending {
+
+    var bonuses: [Bonus]
+
+    func sendRequestForRequestTemplate<T : RequestTemplateProtocol>(requestTemplate: T, completionBlock: ((Result<T.ResultType>) -> Void)?) {
+        if let dateStringForBonusesRequestTemplate = requestTemplate.method.queryParameters["end_time"] {
+            //bonuses request
+            for bonus in bonuses {
+                let dateFormatter = NSDateFormatter.init()
```
Comment (michallaskowski) [inline GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift]: `NSDateFormatter()`?

File: GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    func testWhetherNextPagesAreFetchedWhenFirstOneDoesntContain10UniquePeople() {
+        let expectation = expectationWithDescription("request")
+        var fetchedPeople: [Person]?
+
+        fetchController.fetchPeopleWithBonuses { result in
+            switch result {
+            case .Success(let people):
+                fetchedPeople = people
+            case .Failure(_):
```
Comment (michallaskowski) [inline GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift]: can be just `case .Failure:`

**Comment by michallaskowski** [comment]
That's all :)

File: GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift (hunk 2 above)
Comment (ochococo) [inline GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift]: definitely `forEach`

File: GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift (hunk 3 above)
Comment (ochococo) [inline GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift]: I'm not in favor of strongSelf entirely. 

If owner of mine is no longer interested in me being alive, he most definitely want me dead and doing some calls (including completion) after my supposed dealloc is spooky and complicates things.

File: GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift (hunk 5 above)
Comment (ochococo) [inline GrandCentralBoardTests/PeopleWithBonusesFetchControllerTests.swift]: Fixed.

**Comment by ochococo** [comment]
Back to you @michallaskowski 🎤

**Comment by michallaskowski** [comment]
Yay! Happy to merge it 🔥

**Pull request merged**